src/compiler/core.ts

```diff
@@ -92,6 +92,41 @@ module ts {
         return result;
     }
 
+    export function reduce<T>(array: T[], f: (a: T, x: T) => T): T;

```

Comment:
I was keeping the name consistent with the `Array.prototype` member in ES5, but it doesn't much matter to me.